python/ray/serve/_private/cluster_node_info_cache.py

```diff
@@ -26,25 +26,21 @@ def update(self):
         """
         nodes = self._gcs_client.get_all_node_info(timeout=RAY_GCS_RPC_TIMEOUT_S)
         alive_nodes = [
-            (ray.NodeID.from_binary(node_id).hex(), node["node_name"].decode("utf-8"))
+            (node_id.hex(), node.node_name)
             for (node_id, node) in nodes.items()
-            if node["state"] == ray.core.generated.gcs_pb2.GcsNodeInfo.ALIVE
+            if node.state == ray.core.generated.gcs_pb2.GcsNodeInfo.ALIVE
         ]
 
         # Sort on NodeID to ensure the ordering is deterministic across the cluster.
         sorted(alive_nodes)
         self._cached_alive_nodes = alive_nodes
         self._cached_node_labels = {
-            ray.NodeID.from_binary(node_id).hex(): {
-                label_name.decode("utf-8"): label_value.decode("utf-8")
-                for label_name, label_value in node["labels"].items()
-            }
-            for (node_id, node) in nodes.items()
+            node_id.hex(): dict(node.labels) for (node_id, node) in nodes.items()
```

Comment:
do we need decode here?

Reply:
no need. labels is `map<string, string>` so it's already decoded to python `str`. only proto `bytes` go to python `bytes` and needs decode.